src/utils/toc.ts

```diff
@@ -0,0 +1,41 @@
+import slugger from 'github-slugger';
+
+interface ToCItem {
+  text: string;
+  id: string;
+  level: string;
+  subItems: ToCItem[];
+}
+
+export type TableOfContents = ToCItem[];
+
+export function getTableOfContents(mdxContent: string) {
+  const regexp = new RegExp(/^(### |## )(.*)\n/, 'gm');
```

Comment:
We should handle every headings.

I think it is good to provide `h1` too, it's the Developer responsibility to render it or not in its ToC. Note that there might be several `h1` in the markdown file depending on author's preferences / concerns.

```suggestion
  const regexp = new RegExp(/^(# |## |### |#### |##### |######)(.*)\n/, 'gm'); // h1 -> h6
```